src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java

```diff
@@ -358,14 +361,13 @@ public void write(String text) {
 					if (this.connection.rendezvousWebSocket != null) {
 						flushReason = FlushReason.RENDEZVOUS_EXISTS;
 					} else {
-						int bufferedCount = this.writeBufferStream != null ? this.writeBufferStream.position() : 0;
+						int bufferedCount = this.writeBufferStream != null ? this.writeBufferStream.size() : 0;
 						if (count + bufferedCount <= MAX_CONTROL_CONNECTION_BODY_SIZE) {
 
 							// There's still a chance we might be able to respond over the control
 							// connection, accumulate bytes
 							if (this.writeBufferStream == null) {
-								int initialStreamSize = Math.min(count, MAX_CONTROL_CONNECTION_BODY_SIZE);
```

Comment:
>int initialStreamSize = Math.min(count, MAX_CONTROL_CONNECTION_BODY_SIZE); [](start = 8, length = 74)

Don't think we need to do this min check, "count" won't be greater than MAX_CONTROL_CONNECTION_BODY_SIZE if we got here

Reply:
Good catch.

---
In reply to: [275563868](https://github.com/Azure/azure-relay-java/pull/39#discussion_r275563868) [](ancestors = 275563868)